packages/common/src/utils/internal/resolveDynamicValues.ts

```diff
@@ -0,0 +1,28 @@
+import { IArcGISContext } from "../../ArcGISContext";
+import {
+  DynamicValueDefinition,
+  DynamicValueResult,
+} from "../../core/types/DynamicValues";
+import { resolveDynamicValue } from "./resolveDynamicValue";
+
+/**
+ * @internal
+ * Resolve a set of dynamic values.
+ * The individual definitions can be of many types (item-query, portal etc), each of which has their own logic for resolving the value.
+ * @param dynamicValues Array of dynamic value definitions to resolve
+ * @param context ArcGIS context of the current user.
+ * @returns
+ */
+export const resolveDynamicValues = async (
+  dynamicValues: DynamicValueDefinition[],
+  context: IArcGISContext
+): Promise<Record<string, DynamicValueResult>> => {
+  let result: Record<string, any> = {};
+
+  for (const valueDef of dynamicValues) {
+    const val = await resolveDynamicValue(valueDef, context);
+    result = { ...result, ...val };
+  }
```

Comment:
i assume you have a reason not to use `Promise.all()` - if so maybe document w/ a comment?

Reply:
tbh - I don't have a solid reason for not parallelizing this... just did not want to mess with it b/c it works :) 
I'll re-assess and either use Promise.all or leave a comment